nixos/modules/services/networking/peroxide.nix

```diff
@@ -0,0 +1,180 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.peroxide;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  options.services.peroxide = {
+    enable = mkEnableOption "peroxide service";
+
+    stateDir = mkOption {
+      type = types.str;
+      default = "/var/lib/peroxide";
+      description = ''
+        The state directory, where the cookie-jar, certs, credential store, and
+        log files live.
+      '';
+    };
+
+    selfSignedCert = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Whether to generate a self-signed X.509 certificate on first run.
+
+          If this option is disabled, then you must provide your own certificate
+          and place it in the directory set by
+          <xref linked="opt-services.peroxide.stateDir"/>
+          for the peroxide service to start.
+        '';
+      };
+
+      organization = mkOption {
+        type = types.str;
+        default = "N/A";
+        description = ''
+          Value for the 'Organization' field of the self-signed X.509 certificate.
+        '';
+      };
+
+      hostname = mkOption {
+        type = types.str;
+        default = config.networking.hostName;
+        description = ''
+          Value for the 'CN' field of the self-signed X.509 certificate.
+        '';
+      };
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "peroxide";
+      description = "User as which peroxide is run.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "peroxide";
+      description = "Group as which peroxide is run.";
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          CacheDir = mkOption {
+            type = types.path;
+            default = "${cfg.stateDir}/cache";
+            # See comment on module assertions for an explanation of this
+            # restriction.
+            description = ''
+              The cache directory for peroxide.
+
+              <warning>
+                This <emphasis>must</emphasis> be a subdirectory of
+                <xref linked="opt-services.peroxide.stateDir"/>
+              </warning>
+            '';
+          };
```

Comment:
I don't see the usecase for this option tbh.

Reply:
Me neither, now that you mentioned it.  We still need to set the `CacheDir` peroxide setting because the default won't work on NixOS, but I have moved it to outside the user-configurable options.